flutter_vlc_player/android/src/main/java/software/solid/fluttervlcplayer/FlutterVlcPlayer.java

```diff
@@ -0,0 +1,612 @@
+                                break;
+
+                            case MediaPlayer.Event.Buffering:
+                            case MediaPlayer.Event.TimeChanged:
+                                eventObject.put("event", "timeChanged");
+                                eventObject.put("position", mediaPlayer.getTime());
+                                eventObject.put("speed", mediaPlayer.getRate());
+                                eventObject.put("buffer", event.getBuffering());
+                                mediaEventSink.success(eventObject);
+                                break;
+
+                            case MediaPlayer.Event.EncounteredError:
+                                //mediaEventSink.error("500", "Player State got an error.", null);
+                                eventObject.put("event", "error");
+                                mediaEventSink.success(eventObject);
+                                break;
+
+                            case MediaPlayer.Event.LengthChanged:
+                            case MediaPlayer.Event.MediaChanged:
+                            case MediaPlayer.Event.ESAdded:
+                            case MediaPlayer.Event.ESDeleted:
+                            case MediaPlayer.Event.ESSelected:
+                            case MediaPlayer.Event.PausableChanged:
+                            case MediaPlayer.Event.RecordChanged:
+                            case MediaPlayer.Event.SeekableChanged:
+                            case MediaPlayer.Event.PositionChanged:
+                            default:
+                                break;
+                        }
+                    }
+                }
+        );
+    }
+
+    void play() {
+        mediaPlayer.play();
+    }
+
+    void pause() {
+        mediaPlayer.pause();
+    }
+
+    void stop() {
+        mediaPlayer.stop();
+    }
+
+    boolean isPlaying() {
+        return mediaPlayer.isPlaying();
+    }
+
+    void setStreamUrl(String url, boolean isAssetUrl, boolean autoPlay, long hwAcc) {
+        try {
+            mediaPlayer.stop();
+            //
+            Media media;
+            if (isAssetUrl)
+                media = new Media(libVLC, context.getAssets().openFd(url));
+            else
+                media = new Media(libVLC, Uri.parse(url));
+            if (hwAcc != HwAcc.AUTOMATIC.getNumericType()) {
+                if (hwAcc == HwAcc.DISABLED.getNumericType()) {
+                    media.setHWDecoderEnabled(false, false);
+                } else if (hwAcc == HwAcc.FULL.getNumericType() || hwAcc == HwAcc.DECODING.getNumericType()) {
+                    media.setHWDecoderEnabled(true, true);
+                    if (hwAcc == HwAcc.DECODING.getNumericType()) {
+                        media.addOption(":no-mediacodec-dr");
+                        media.addOption(":no-omxil-dr");
+                    }
+                }
+            }
```

Comment:
Nitpick: seems a bit convoluted. Could we turn it into a `switch`, perhaps?
```suggestion
            final HwAcc hwAcc = HwAcc.fromValue(hwAccValue);
            switch (hwAcc) {
                case DISABLED:
                    media.setHWDecoderEnabled(false, false);
                    break;
                case DECODING:
                case FULL:
                    media.setHWDecoderEnabled(true, true);
                    break;
            }
            if (hwAcc == HwAcc.DECODING) {
                media.addOption(":no-mediacodec-dr");
                media.addOption(":no-omxil-dr");
            }
```